#**Ratio de Sharpe**
El ratio de Sharpe es una medida utilizada en finanzas para evaluar el rendimiento de una inversión en relación con su riesgo. Fue desarrollado por el premio Nobel William F. Sharpe y se calcula dividiendo el exceso de rendimiento de la inversión (es decir, el rendimiento de la inversión menos la tasa libre de riesgo) entre la volatilidad o desviación estándar de los rendimientos de la inversión.

## **Calculo del Ratio de Sharpe**


$$ S = {\frac {Rp-Rf}{\sigma}}$$

Donde:
- **(Rp)** = Rentabilidad del Portafolio
- **(Rf)** = Tasa Libre de Riesgo
- **$\sigma$** = Desviación Estandar

## **Interpretación**

- Un ratio de Sharpe alto (>1) indica que la inversión ha generado un buen rendimiento en relación con el riesgo asumido.

- Un ratio de Sharpe bajo (<1) sugiere que el rendimiento no compensa adecuadamente el riesgo tomado.

- Un ratio de Sharpe negativo indica que la inversión ha tenido un rendimiento inferior a la tasa libre de riesgo.

##**EJERCICIO APLICATIVO**
La empresa "FinanzasPro S.A." está evaluando dos opciones de inversión para el próximo año:

####Inversión TechGrowth####
Un fondo de acciones tecnológicas con un rendimiento esperado del 15%, una volatilidad del 20% y una tasa libre de riesgo del 5%.

####Inversión SafeBonds####
Un fondo de bonos gubernamentales con un rendimiento esperado del 8%, una volatilidad del 5% y la misma tasa libre de riesgo del 5%.

**La empresa quiere comparar ambas inversiones utilizando el ratio de Sharpe para decidir cuál es la mejor opción.**



In [ ]:
# Variables de la Inversión TechGrowth
rendimiento_tech = 0.15  # Rendimiento esperado del 15%
volatilidad_tech = 0.20  # Volatilidad del 20%

# Variables de la Inversión SafeBonds
rendimiento_bonds = 0.08  # Rendimiento esperado del 8%
volatilidad_bonds = 0.05   # Volatilidad del 5%

# Tasa libre de riesgo (común para ambas inversiones)
tasa_libre_riesgo = 0.05  # Tasa libre de riesgo del 5%

In [ ]:
# Cálculo del ratio de Sharpe para la Inversión TechGrowth
sharpe_tech = (rendimiento_tech - tasa_libre_riesgo) / volatilidad_tech

# Cálculo del ratio de Sharpe para la Inversión SafeBonds
sharpe_bonds = (rendimiento_bonds - tasa_libre_riesgo) / volatilidad_bonds

# Mostrar los resultados
print(f"Ratio de Sharpe de la Inversión TechGrowth: {sharpe_tech:.2f}")
print(f"Ratio de Sharpe de la Inversión SafeBonds: {sharpe_bonds:.2f}")

Ratio de Sharpe de la Inversión TechGrowth: 0.50
Ratio de Sharpe de la Inversión SafeBonds: 0.60


In [ ]:
# Comparación de los ratios de Sharpe
if sharpe_tech > sharpe_bonds:
    print("La Inversión TechGrowth es mejor que la Inversión SafeBonds según el ratio de Sharpe.")
elif sharpe_bonds > sharpe_tech:
    print("La Inversión SafeBonds es mejor que la Inversión TechGrowth según el ratio de Sharpe.")
else:
    print("Ambas inversiones tienen el mismo ratio de Sharpe.")

# Interpretación adicional
print("\nInterpretación:")
print("- Un ratio de Sharpe más alto indica un mejor rendimiento ajustado al riesgo.")
print("- La inversión con el ratio más alto es preferible, ya que ofrece mayor retorno por unidad de riesgo.")

La Inversión SafeBonds es mejor que la Inversión TechGrowth según el ratio de Sharpe.

Interpretación:
- Un ratio de Sharpe más alto indica un mejor rendimiento ajustado al riesgo.
- La inversión con el ratio más alto es preferible, ya que ofrece mayor retorno por unidad de riesgo.


##Conclusión##
Explicación del resultado:
Aunque la Inversión TechGrowth tiene un rendimiento esperado más alto (15% vs 8%), su volatilidad es mucho mayor (20% vs 5%).

La Inversión SafeBonds, aunque tiene un rendimiento más bajo, tiene un ratio de Sharpe más alto (0.60 vs 0.50), lo que indica que ofrece un mejor rendimiento por unidad de riesgo asumido.

####**Por lo tanto, FinanzasPro S.A. debería elegir la Inversión SafeBonds.**####